integration/helpers/archive.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package helpers
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+)
+
+// CompressDirToZipFile compresses directory into a `.zip` format.
+func CompressDirToZipFile(sourcePath, destPath string) error {
+	archive, err := os.Create(destPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
```

Comment:
```suggestion
defer archive.Close()
```

Then you don't need `_ = archive.Close()` in the error branches. Same for CompressDirToTarGzFile.